app/repos/people.js

```diff
@@ -303,6 +301,95 @@ const updatePersonFields = async (id, personFields, user, transaction) => {
 
   return person
 }
+/**
+ * @typedef RegisteredPerson
+ * @property {number} id
+ * @property {number} person_id
+ * @property {number} dog_id
+ * @property {number} person_type_id
+ * @property {DogDao} dog
+ */
+/**
+ * @typedef {unknown} PersonContact
+ */
+/**
+ * @typedef PersonWithRegisteredPeople
+ * @property {number} id
+ * @property {string} first_name
+ * @property {string} last_name
+ * @property {string} person_reference
+ * @property {string} birth_date
+ * @property {PersonAddressDao[]} addresses
+ * @property {PersonContact[]} person_contacts
+ * @property {RegisteredPerson[]} registered_people
+ */
+
+/**
+ * @typedef PersonAndDogsByIndexDao
+ * @property {number} id
+ * @property {number} person_id
+ * @property {number} dog_id
+ * @property {number} person_type_id
+ * @property {DogDao} dog
+ * @property {PersonWithRegisteredPeople} person
+ */
+
+/**
+ * @param {string} indexNumber
+ * @param [transaction]
+ * @return {Promise<PersonAndDogsByIndexDao>}
+ */
+const getPersonAndDogsByIndex = async (indexNumber, transaction) => {
+  try {
+    return await sequelize.models.registered_person.findOne({
+      include: [
```

Comment:
I think you will need the address orderBy here otherwise there is a danger you will receive an old address when you choose address[0]